service/state/core.go

```diff
@@ -88,28 +59,14 @@ func NewCore(dir string) (*Core, error) {
 	err = db.View(func(tx *badger.Txn) error {
 
 		// first we get the last commit
-		key := []byte{prefixLastCommit}
-		item, err := tx.Get(key)
-		if err != nil {
+		if err := RetrieveLastCommit(&commit)(tx); err != nil {
 			return fmt.Errorf("could not retrieve last commit: %w", err)
 		}
-		_ = item.Value(func(val []byte) error {
-			commit = val
-			return nil
-		})
 
 		// then we get the height associated with it
-		key = make([]byte, 1+len(commit))
-		key[0] = prefixIndexCommit
-		copy(key[1:], commit)
-		item, err = tx.Get(key)
-		if err != nil {
-			return fmt.Errorf("could not access commit index: %w", err)
+		if err := RetrieveCommitHeight(commit, &height)(tx); err != nil {
```

Comment:
I would suggest alternative naming for better readability here: `RetrieveHeightByCommit`. We can rename the prefixes/indexes as well to avoid confusion: `prefixCommitToHeight`.